### PR TITLE
Feat/user profile in townhall

### PR DESCRIPTION
**Pull request by Hitesh-gurnani**

### CODE REVIEW

**Comment by This-Is-Prince** [comment]
I have created more than 50 rooms, but in my profile i'm only able to see 9. There is no modal which can show all the rooms with pagination.

[multiple-created-rooms.webm](https://github.com/townhall-gov/townhall/assets/89686019/5df73f48-a804-4ee9-ace9-0b99e75260c0)


Screenshot 2023-09-20 at 7 12 49 AM

**Comment by This-Is-Prince** [comment]
Getting this error when connecting second account and i'm still in my previous profile page.

Screenshot 2023-09-20 at 7 17 53 AM

Screenshot 2023-09-20 at 7 17 53 AM (2)

**Comment by This-Is-Prince** [comment]
# There is no proper testing done for this PR

[editprofile-bug.webm](https://github.com/townhall-gov/townhall/assets/89686019/2dbb68c8-7244-45af-81fe-0d58159b622b)

File: src/ui-components/TextEditor/index.tsx [hunk 1]
```diff
@@ -328,4 +328,4 @@ export default styled(TextEditor)`
 			scale: 0.85;
 		}
 	}
-`;
\ No newline at end of file
+`;
```
[the remark this answers is not in the document]
Comment (This-Is-Prince) [reply src/ui-components/TextEditor/index.tsx:331]: Still pending.....

**Review by This-Is-Prince — CHANGES_REQUESTED** [review]
Please test feature properly

File: src/ui-components/UserTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+							<Button
+								className={'outline-none border cursor-pointer rounded-full h-full border-solid border-[#66A5FF] flex items-center justify-center bg-blue_primary text-white  max-w-[100px] w-full text-base leading-[19px] font-normal tracking-[0.01em]'}
+							>
+                    View Profile
+							</Button>
+						</Link>
+				}
+				{
+					user_info?.address?.startsWith('0x')?
+						<Link href={`https://etherscan.io/address/${user_info.address}`} target="_blank">
```
[the remark this answers is not in the document]
Comment (Hitesh-gurnani) [reply src/ui-components/UserTooltip.tsx]:
```
const generateProfileButtons = (user: IUserInfo | null) => {
	if (!user) return null;

	const profileLink = user.address ? `/user/${user.address}` : '';
	const explorerLink = user.address?.startsWith('0x')
		? `https://etherscan.io/address/${user.address}`
		: `https://www.subscan.io/account/${user?.address}`;

	return (
		<div className='flex flex-row justify-between items-center w-full pt-3'>
			{profileLink && (
				<Link href={profileLink}>
					<ViewProfileButton />
				</Link>
			)}
			{explorerLink && (
				<Link href={explorerLink} target="_blank">
					<SeeExplorerButton />
				</Link>
			)}
		</div>
	);
};
```

File: src/ui-components/UserTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const UserTooltip: FC<IUserTooltipProps> = (props) => {
+	const { user_info } = props;
+	return (
+		<article className='flex flex-col p-2 min-w-[240px] min-h-[20vh] justify-between items-center rounded-lg'>
+			<div>
+				{
+					user_info?
+						user_info.username?
+							<div className='flex  flex-col px-8 justify-center items-center  w-[240px]'>
```
[the remark this answers is not in the document]
Comment (Hitesh-gurnani) [reply src/ui-components/UserTooltip.tsx]:
```
{user.img_url ? (
						<Image preview={false} width={user.username ? 75 : 90} height={user.username ? 75 : 90} className='rounded-full'
							src={user.img_url} alt={user.name as string} />
					) : (
						<div className='flex items-center'>
							{!disableIdenticon && (
								<div className='flex items-center justify-center'>
									{address.startsWith('0x') ? (
										<EthIdenticon className='flex items-center justify-center' size={60} address={address} />
									) : (
										<Identicon className='flex items-center justify-center' value={address} size={60} theme={'polkadot'} />
									)}
								</div>
							)}
						</div>
					)}
```

**Pull request closed without merging**